core/base/src/TApplication.cxx

```diff
@@ -628,20 +630,423 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
+   } else {
+      // For Linux we check if the DISPLAY is set
+      TString displayhandle(gSystem->Getenv("DISPLAY"));
+      if ((displayhandle.Sizeof()) > 1) {
+         // If the DISPLAY is set it will open the browser
+         cLinux.Append(url);
+         gSystem->Exec(cLinux.Data());
+      } else {
+         // Else the user will have a warning and the URL in the terminal
+         Warning("Help", "The DISPLAY is not set!");
+         std::cout << url.Data() << std::endl;
+      }
+   }
+}
+namespace {
+static TString URLGenerator(TString ClassName)
+{
+   // Generates a URL address for the class (classname).
+   // This is the URL to the online reference guide, generated by Doxygen.
+   TString URLBegining("https://root.cern.ch/doc/master/class");
+   TString URLEnd(".html");
+   URLBegining.Append(ClassName);
+   URLBegining.Append(URLEnd);
+   return URLBegining.Data();
+}
+}
+
+namespace {
+static TString URLforNameSpace(TString NameSpace)
+{
+   // Generates a URL address for the namespace (NameSpace).
+   // This is the URL to the online reference guide, generated by Doxygen.
+   TString URLbegining("https://root.cern/doc/master/namespace");
+   TString URLend(".html");
+   URLbegining.Append(NameSpace.Data());
+   URLbegining.Append(URLend);
+   return URLbegining.Data();
+}
+}
+
+namespace {
+static TString DoxygenSyntax(TString ClassName, TString Method)
+{
+   // The function returns a TString with the arguments of a member function (Method)
+   // of a class (ClassName), but modified with respect to the Doxygen syntax - spacing
+   // around special sumbols and adding the missing sumbols ("std::", ",...").
+   // "DoxygenSyntax" works for the functions defined inside namespaces too.
+   TString MethodArguments(((TMethod *)TClass::GetClass(ClassName)->GetListOfMethods()->FindObject(Method))->GetSignature());
+   // With "GetSignature" we get the arguments of "Method" and put them in a TString.
+   // Then the TString "MethodArguments" is modified.
+   MethodArguments.ReplaceAll(" = ", "=");
+   // maybe add also " =" and "= ";
+   MethodArguments.ReplaceAll("* ", " *");
+   // is this always? below
+   MethodArguments.ReplaceAll("*=", " *=");
+   MethodArguments.ReplaceAll("*)", " *)");
+   MethodArguments.ReplaceAll("*,", " *,");
+   MethodArguments.ReplaceAll("*& ", " *&");
+   MethodArguments.ReplaceAll("& ", " &");
+   MethodArguments.ReplaceAll("ostream", "std::ostream");
+   MethodArguments.ReplaceAll("istream", "std::istream");
+   MethodArguments.ReplaceAll("*msgfmt", "*msgfmt,...");
+   TString Buff ("\\b");
+   Buff.Append(ClassName);
+   Buff.Append("::\\b");
+   TPRegexp ArgFix(Buff.Data());
+   ArgFix.Substitute(MethodArguments, "");
+   return MethodArguments;
+}
+}
+
+namespace {
+static TString 
+InlineFuncCheck(TString ClassName, TString Method)
+{
+   // The function checks if a member function (Method) of a class (ClassName) is
+   // defined as inline. If so, it also checks if it is virtual.
```

Comment:
Why is the virtual check only happening if the function is inline?

Maybe "Adjust return type of the function to consider virtual / inline as required by doxygen hash calculation"?

Reply:
If the function is not inline, Doxygen will not take into account if it's virtual or not. If the function is inline, then the return type must be modified. I didn't change the code here.